.github/workflows/gh-page_builder.yml

```diff
@@ -11,29 +16,35 @@ jobs:
     steps:
     - uses: actions/checkout@v2
     - name: Set up Python
-      uses: actions/setup-python@v1
+      uses: actions/setup-python@v4
       with:
         python-version: 3.7 
-    - name: Install Python dependencies
+
+    - name: Install dependencies
       run: |
-        sudo apt-get install python3-pip
-        #pip install git+git://github.com/executablebookproject/cli.git#egg=master
-        pip install ghp-import
-        #pip install -r book/requirements.txt
+        python -m pip install --upgrade pip
         pip install jupyter-book
         pip install sphinx-pythia-theme
+        pip install -r requirements.txt
         PATH="${PATH}:${HOME}/.local/bin"
    # - name: Build book TOC file 
    #   run: |
    #     #jupyter-book toc .
     - name: Build book HTML 
       run: |
         jupyter-book build .
-    - name: Push _build/html to gh-pages
-      run: |
-        sudo chown -R $(whoami):$(whoami) .
-        git config --global user.email "$GITHUB_ACTOR@users.noreply.github.com"
-        git config --global user.name "$GITHUB_ACTOR"
-        git remote set-url origin "https://$GITHUB_ACTOR:${{ secrets.GITHUB_TOKEN }}@github.com/$GITHUB_REPOSITORY"
-        ls -lrt
-        ghp-import _build/html -f -p -n
+    - name: Upload artifact
+      uses: actions/upload-pages-artifact@v3
+      with:
+        path: ./_build/html
```

Comment:
@halehawk do we need this step?

Reply:
@negin513 Yes, this step can zip all the html files and upload to the website.